modules/cbQuestion/actions/qactions.php

```diff
@@ -301,5 +301,24 @@ public function delMViewWF() {
 		$rdo['notify'] = $smarty->fetch('applicationmessage.tpl');
 		echo json_encode($rdo);
 	}
+
+	public function executeScript() {
+		$record = $this->checkQIDParam();
+		$dbUser = isset($_REQUEST['db_user']) ? vtlib_purify($_REQUEST['db_user']) : '';
+		$dbPass = isset($_REQUEST['db_pass']) ? vtlib_purify($_REQUEST['db_pass']) : '';
+		$dbName = isset($_REQUEST['db_name']) ? vtlib_purify($_REQUEST['db_name']) : '';
+		$tablename = isset($_REQUEST['tablename']) ? vtlib_purify($_REQUEST['tablename']) : '';
+		$scriptPath = isset($_REQUEST['script_path']) ? vtlib_purify($_REQUEST['script_path']) : '';
+
+		$command = sprintf("".$scriptPath." %s %s %s %s %s", $dbUser, $dbPass, $dbName, $tablename, $record);
+		$output = shell_exec($command);
```

Comment:
we have to add a list of authorized scripts here as this is to dangerous to leave open like this. I propose a simple json/xml or a database table with the list of authorized scripts